src/core/common/message.cpp

```diff
@@ -99,15 +100,20 @@ void MessagePool::Free(Message *aMessage)
     FreeBuffers(static_cast<Buffer *>(aMessage));
 }
 
-Buffer *MessagePool::NewBuffer(void)
+Buffer *MessagePool::NewBuffer(uint8_t aPriority)
 {
     Buffer *buffer = NULL;
 
 #if OPENTHREAD_CONFIG_PLATFORM_MESSAGE_MANAGEMENT
 
     buffer = static_cast<Buffer *>(otPlatMessagePoolNew(&GetInstance()));
+    OT_UNUSED_VARIABLE(aPriority);
 
 #else
+    if (mFreeBuffers == NULL)
+    {
+        ReclaimBuffers(1, aPriority);
```

Comment:
When `OPENTHREAD_CONFIG_PLATFORM_MESSAGE_MANAGEMENT` is `1`, we should be calling `ReclaimBuffers()` when there are no buffers available.  Do we need to move `ReclaimBuffers()` out of `MessagePool::New()` to here?

Reply:
The function ReclaimBuffers() should be moved to the head of the function NewBuffer(). Then, when OPENTHREAD_CONFIG_PLATFORM_MESSAGE_MANAGEMENT is 1, it still works.  The function Prepend() and ResizeMessage() call NewBuffer() to malloc buffers with priority, so we need to move ReclaimBuffers() out of MessagePool::New() to NewBuffer().